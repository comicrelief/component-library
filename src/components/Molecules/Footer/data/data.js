export default {
  menuDescription: 'Footer',
  menuGroup: [
    {
      id: 'f7dc5eef-f4eb-5405-8a35-08808b55cb33',
      title: 'Hear from us',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Get the newsletter',
          url: 'https://www.comicrelief.com/#',
          class: null
        },
        {
          title: 'How we contact you',
          url: 'https://www.comicrelief.com/update-your-preferences',
          class: null
        }
      ]
    },
    {
      id: '9f3980f3-d02b-52e9-ac41-933a778c040a',
      title: 'Get in touch',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Contact us',
          url: 'https://www.comicrelief.com/contact-us',
          class: null
        },
        {
          title: 'Your Gift Aid',
          url: 'https://giftaid.comicrelief.com/update',
          class: null
        },
        {
          title: 'FAQs',
          url: 'https://www.comicrelief.com/frequently-asked-questions',
          class: null
        },
        {
          title: 'Red Nose Day 2019 FAQs',
          url:
            'https://www.comicrelief.com/red-nose-days-frequently-asked-questions',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998b',
      title: 'About us',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Mission',
          url: 'https://www.comicrelief.com/about-comic-relief',
          class: null
        },
        {
          title: 'Our history',
          url: 'https://www.comicrelief.com/about-comic-relief/history',
          class: null
        },
        {
          title: 'Finances',
          url: 'https://www.comicrelief.com/about-comic-relief/finances',
          class: null
        },
        {
          title: 'Partners',
          url: 'https://www.comicrelief.com/working-with-us',
          class: null
        },
        {
          title: 'Meet the team',
          url: 'https://www.comicrelief.com/about-comic-relief/meet-the-team',
          class: null
        },
        {
          title: 'Sport Relief',
          url: 'https://www.sportrelief.com/',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373c',
      title: 'Careers',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'Working at Comic Relief',
          url: 'https://www.comicrelief.com/careers',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://app.beapplied.com/org/comic-relief',
          class: null
        }
      ]
    },
    {
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373d',
      title: 'News',
      url: 'https://www.comicrelief.com',
      links: [
        {
          title: 'News',
          url: 'https://www.comicrelief.com/news',
          class: null
        },
        {
          title: 'Open rolls',
          url: 'https://www.comicrelief.com/press-releases',
          class: null
        }
      ]
    },
    {
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998a',
      title: 'Legal',
      url: 'https://www.comicrelief.com/',
      links: [
        {
          title: 'Privacy policy',
          url: 'https://www.comicrelief.com/privacy-notice',
          class: null
        },
        {
          title: 'Terms of use',
          url: 'https://www.comicrelief.com/terms-of-use',
          class: null
        },
        {
          title: 'Cookies',
          url: 'https://www.comicrelief.com/cookies-policy',
          class: null
        },
        {
          title: 'Positive practices',
          url: 'https://www.comicrelief.com/positive-practices',
          class: null
        },
        {
          title: 'Text to donate terms of use',
          url: 'https://www.comicrelief.com/legal/sms',
          class: null
        }
      ]
    }
  ]
};
