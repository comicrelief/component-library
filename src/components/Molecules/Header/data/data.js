export default {
  menuDescription: 'I am the navigation',
  menuGroup: [
    {
      id: 'Parent Item Title 1',
      title: 'Parent Item Title 1',
      url: '/parent1',
      links: [
        {
          title: 'Child Item Title 1-1',
          url: '/child-item-1-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-2',
          url: '/child-item-1-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 1-3',
          url: '/child-item-1-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 2',
      title: 'Parent Item Title 2',
      url: '/parent2',
      links: [
        {
          title: 'Child Item Title 2-1',
          url: '/child-item-2-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-2',
          url: '/child-item-2-2',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 2-3',
          url: '/child-item-2-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 4',
      title: 'Parent Item Title 4',
      url: '/parent4',
      links: [
        {
          title: 'Child Item Title 4-1',
          url: '/child-item-4-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-4',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 4-4',
          url: '/child-item-4-3',
          class: 'someClass'
        }
      ]
    },
    {
      id: 'Parent Item Title 3',
      title: 'Parent Item Title 3',
      url: '/parent3',
      links: [
        {
          title: 'Child Item Title 3-1',
          url: '/child-item-3-1',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        },
        {
          title: 'Child Item Title 3-3',
          url: '/child-item-3-3',
          class: 'someClass'
        }
      ]
    }
  ]
};
