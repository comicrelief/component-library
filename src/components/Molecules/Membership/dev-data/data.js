export default {
  cartId: 'test-sept2019-C365',
  regularGiving: {
    otherDescription:
      'Your donation will help us fund amazing projects in the UK and around the world.',
    moneybuys: [
      {
        amount: 10,
        description:
          'a regular supply of toiletries for someone living in a refugee camp in Serbia.'
      },
      {
        amount: 20,
        description:
          'a potentially lifesaving call for a man at risk of suicide in the UK.'
      },
      {
        amount: 30,
        description:
          'the distribution of enough surplus food for a school breakfast club to feed 80 children.'
      }
    ]
  }
};
