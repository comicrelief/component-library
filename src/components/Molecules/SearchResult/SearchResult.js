import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Text from '../../Atoms/Text/Text';
import Picture from '../../Atoms/Picture/Picture';
import Link from '../../Atoms/Link/Link';
import spacing from '../../../theme/shared/spacing';

/**
 * Search Result
 */
const Wrapper = styled.article`
  width: 100%;
  height: 100%;
  background-color: ${({ theme }) => theme.color('white')};
`;

const Item = styled(Link)`
  display: flex;
  height: 100%;
  border: 0;
  flex-direction: column;
  text-decoration: none;
  color: inherit;
  width: 100%;
  @media ${({ theme }) => theme.allBreakpoints('M')} {
    flex-direction: row;
  }
  :hover {
    border: 0;
    font-weight: normal;
  }
`;

const ImageWrapper = styled.div`
  height: auto;
  @media ${({ theme }) => theme.allBreakpoints('M')} {
    width: ${({ smallImageWidth }) => smallImageWidth};
  }
  @media ${({ theme }) => theme.allBreakpoints('XL')} {
    width: ${({ largeImageWidth }) => largeImageWidth};
  }
`;

const CopyWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: ${spacing('md')};
  width: 100%;
`;

const Title = styled(Text)`
  margin: 0;
`;

/**
 * Search Result component
 */
const SearchResult = ({
  imageLow = null,
  images = null,
  type = '',
  date = '',
  copy = '',
  smallImageWidth = '45%',
  largeImageWidth = '100%',
  href,
  title,
  alt
}) => (
  <Wrapper>
    <Item href={href} type="standard">
      <ImageWrapper
        smallImageWidth={smallImageWidth}
        largeImageWidth={largeImageWidth}
      >
        <Picture
          imageLow={imageLow}
          images={images}
          objectFit="cover"
          alt={alt}
        />
      </ImageWrapper>
      <CopyWrapper>
        <Text size="xs" uppercase>
          {`${type ? ` ${type} |` : ''} ${date}`}
        </Text>
        <Title size="xl" tag="h3" uppercase>
          {title}
        </Title>
        {copy && (
        <Text size="m" tag="p">
          {copy}
        </Text>
        )}
      </CopyWrapper>
    </Item>
  </Wrapper>
);

SearchResult.propTypes = {
  /** type */
  type: PropTypes.string,
  /** copy */
  copy: PropTypes.string,
  /** date */
  date: PropTypes.string,
  /** image width on small screens */
  smallImageWidth: PropTypes.string,
  /** image width on large screens */
  largeImageWidth: PropTypes.string,
  /** image alt */
  alt: PropTypes.string.isRequired,
  /** title */
  title: PropTypes.string.isRequired,
  /** link url */
  href: PropTypes.string.isRequired,
  /** image url */
  images: PropTypes.string,
  imageLow: PropTypes.string
};

export default SearchResult;
