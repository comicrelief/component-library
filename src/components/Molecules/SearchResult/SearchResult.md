### Minimalist

```js
const defaultData = require('../../../styleguide/data/data').defaultData;
<SearchResult
  href="/test"
  imageLow={defaultData.imageLow}
  images={defaultData.images}
  title="News article"
  alt="Image's description"
/>;
```

### With copy

```js
const defaultData = require('../../../styleguide/data/data').defaultData;
<SearchResult
  href="/test"
  imageLow={defaultData.imageLow}
  images={defaultData.images}
  title="News article"
  copy="News article copy"
  alt="Image's description"
/>;
```

### With date

```js
const defaultData = require('../../../styleguide/data/data').defaultData;
<SearchResult
  href="/test"
  imageLow={defaultData.imageLow}
  images={defaultData.images}
  date="01 July 2019"
  title="News article"
  alt="Image's description"
/>;
```

### With date and type

```js
const defaultData = require('../../../styleguide/data/data').defaultData;
<SearchResult
  href="/test"
  imageLow={defaultData.imageLow}
  images={defaultData.images}
  date="01 July 2019"
  type="article"
  title="News article"
  alt="Image's description"
/>;
```
