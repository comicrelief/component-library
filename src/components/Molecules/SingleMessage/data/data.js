export default {
  title: 'What your money does',
  text:
    'When you donate to Comic Relief or Sport Relief, youâ€™re supporting vulnerable people and communities in the UK and internationally.',
  pictures: {
    small: 'src/components/Atoms/Picture/assets/picture-1.jpg',
    medium: 'src/components/Atoms/Picture/assets/picture-2.jpg',
    large: 'src/components/Atoms/Picture/assets/picture-3.jpg'
  },
  ctaText: 'Find out how'
};
