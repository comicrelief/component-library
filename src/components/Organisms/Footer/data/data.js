export default {
  title: 'Footer',
  menuGroups: [
    {
      title: 'Hear from us',
      id: 'f7dc5eef-f4eb-5405-8a35-08808b55cb33',
      links: [
        {
          title: 'Get the newsletter',
          path: 'https://www.comicrelief.com/#',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'How we contact you',
          path: 'https://www.comicrelief.com/update-your-preferences',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Get in touch',
      id: '9f3980f3-d02b-52e9-ac41-933a778c040a',
      links: [
        {
          title: 'Contact us',
          path: 'contact-us',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Your Gift Aid',
          path: 'https://giftaid.comicrelief.com/update',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'FAQs',
          path: 'https://www.comicrelief.com/frequently-asked-questions',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Red Nose Day 2019 FAQs',
          path:
            'https://www.comicrelief.com/red-nose-days-frequently-asked-questions',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'About us',
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998b',
      links: [
        {
          title: 'Mission',
          path: 'https://www.comicrelief.com/about-comic-relief',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Our history',
          path: 'https://www.comicrelief.com/about-comic-relief/history',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Finances',
          path: 'https://www.comicrelief.com/about-comic-relief/finances',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Partners',
          path: 'https://www.comicrelief.com/working-with-us',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Meet the team',
          path: 'https://www.comicrelief.com/about-comic-relief/meet-the-team',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Sport Relief',
          path: 'https://www.sportrelief.com/',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Careers',
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373c',
      links: [
        {
          title: 'Working at Comic Relief',
          path: 'https://www.comicrelief.com/careers',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Open roles',
          path: 'https://app.beapplied.com/org/comic-relief',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'News',
      id: '4a5b055f-1f7e-5f4b-827d-768032fd373d',
      links: [
        {
          title: 'News',
          path: 'https://www.comicrelief.com/news',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Press area',
          path: 'https://www.comicrelief.com/press-releases',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Legal',
      id: 'eaec5191-bbb3-5e8d-b966-c53fff34998a',
      links: [
        {
          title: 'Link comp with both URL and Ref',
          url: 'https://www.google.com',
          reference: {
            path: 'homepage'
          },
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Link comp with only Ref',
          url: null,
          reference: {
            path: 'this-is-a-test'
          },
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Link comp with only URL',
          url: 'https://www.yahoo.com',
          reference: null,
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Test whitelisted external link',
          url: 'https://www.sportrelief.com',
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Test non-whitelisted external link',
          url: 'https://bing.com',
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Test non-whitelisted external link',
          url: 'https://bing.com',
          internal: {
            type: 'ContentfulComponentLink'
          }
        }
      ]
    }
  ]
};
