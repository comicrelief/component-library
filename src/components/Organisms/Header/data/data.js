export default {
  title: 'Header',
  menuGroups: [
    {
      title: 'Fundraising - menu group',
      id: 'fundraising-menu-group',
      links: [
        {
          title: 'Fundraising',
          path: 'https://www.comicrelief.com/fundraising/pay-in-your-money',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Red Nose Day',
          path: 'https://www.comicrelief.com/rednoseday',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Regular donations',
          path: 'https://www.comicrelief.com/join',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Squads',
          path: 'https://www.comicrelief.com/squads',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'The Noseys',
          path: 'https://www.comicrelief.com/rednoseday/fundraising/the-noseys',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Free downloads',
          path:
            'https://www.comicrelief.com/rednoseday/fundraising/free-downloads',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Single Page - menu group',
      id: 'single-page-menu-group',
      links: [
        {
          title: 'Single menu link',
          path: 'https://www.comicrelief.com/rednoseday/schools',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'What your money does - menu group',
      id: 'wymd-menu-group',
      links: [
        {
          title: 'What your money does',
          path: 'https://www.comicrelief.com/what-your-money-does',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Our legacy',
          path: 'https://www.comicrelief.com/our-legacy',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'Schools & youth - menu group',
      id: 'schools-menu-group',
      links: [
        {
          title: 'Schools & youth (MORE NAV)',
          path: 'https://www.comicrelief.com/rednoseday/schools',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Primary schools (MORE NAV)',
          path:
            'https://www.comicrelief.com/rednoseday/schools/primary-schools',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Secondary schools (MORE NAV)',
          path:
            'https://www.comicrelief.com/rednoseday/schools/secondary-schools',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Nurseries (MORE NAV)',
          path: 'https://www.comicrelief.com/rednoseday/schools/nurseries',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Youth groups (MORE NAV)',
          path: 'https://www.comicrelief.com/rednoseday/youth',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        },
        {
          title: 'Free downloads (MORE NAV)',
          path: 'https://www.comicrelief.com/rednoseday/schools/free-downloads',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'More Nav single page group (single link MORE NAV)',
      id: 'single-more-menu-group',
      links: [
        {
          title: 'Partners (single link MORE NAV)',
          path: 'https://www.comicrelief.com/working-with-us/',
          internal: {
            type: 'ContentfulPageLandingPage'
          }
        }
      ]
    },
    {
      title: 'More Nav double link (single link MORE NAV)',
      id: 'single-more-menu-group',
      links: [
        {
          title: 'Homepage (MORE NAV)',
          url: 'https://www.comicrelief.com/working-with-us/',
          internal: {
            type: 'ContentfulComponentLink'
          }
        },
        {
          title: 'Something (MORE NAV)',
          url: 'https://www.comicrelief.com/404/',
          internal: {
            type: 'ContentfulComponentLink'
          }
        }
      ]
    }
    // {
    //   title: 'External Links (menu group)',
    //   id: 'external-menu-group',
    //   links: [
    //     {
    //       title: 'Multi-item (MORE NAV)',
    //       url: 'https://www.sportrelief.com',
    //       internal: {
    //         type: 'ContentfulComponentLink'
    //       }
    //     },
    //     {
    //       title: 'item2 (MORE NAV)',
    //       url: 'https://bing.com',
    //       internal: {
    //         type: 'ContentfulComponentLink'
    //       }
    //     },
    //     {
    //       title: 'item3 (MORE NAV)',
    //       url: 'https://www.google.com',
    //       reference: {
    //         path: 'test-ref-path-1'
    //       },
    //       internal: {
    //         type: 'ContentfulComponentLink'
    //       }
    //     },
    //     {
    //       title: 'item4 (MORE NAV)',
    //       url: null,
    //       reference: {
    //         path: 'test-ref-path-2'
    //       },
    //       internal: {
    //         type: 'ContentfulComponentLink'
    //       }
    //     },
    //     {
    //       title: 'item5 (MORE NAV)',
    //       url: 'test-url',
    //       reference: null,
    //       internal: {
    //         type: 'ContentfulComponentLink'
    //       }
    //     }
    //   ]
    // }
  ]
};
