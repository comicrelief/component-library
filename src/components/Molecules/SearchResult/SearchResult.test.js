import React from 'react';
import 'jest-styled-components';
import renderWithTheme from '../../../../tests/hoc/shallowWithTheme';
import SearchResult from './SearchResult';
import { defaultData } from '../../../styleguide/data/data';

it('renders correctly in minimalist form', () => {
  const tree = renderWithTheme(
    <SearchResult
      href="/test"
      image={defaultData.image}
      title="News article"
      alt="Image's description"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with copy', () => {
  const tree = renderWithTheme(
    <SearchResult
      href="/test"
      image={defaultData.image}
      title="News article"
      copy="News article copy"
      alt="Image's description"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with date', () => {
  const tree = renderWithTheme(
    <SearchResult
      href="/test"
      image={defaultData.image}
      date="01 July 2019"
      title="News article"
      alt="Image's description"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly with date and type', () => {
  const tree = renderWithTheme(
    <SearchResult
      href="/test"
      image={defaultData.image}
      date="01 July 2019"
      type="article"
      title="News article"
      alt="Image's description"
    />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
