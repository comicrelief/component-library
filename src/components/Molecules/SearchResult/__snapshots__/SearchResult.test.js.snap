// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly in minimalist form 1`] = `
.c7 {
  font-size: 0.83125rem;
  text-transform: uppercase;
}

.c8 {
  font-size: 1.725rem;
  text-transform: uppercase;
}

.c4 {
  position: relative;
  width: 100%;
  height: auto;
}

.c5 {
  display: block;
  width: 100%;
  height: auto;
  object-fit: cover;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0 2px 1px;
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: normal;
}

.c2:hover {
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: 700;
}

.c0 {
  width: 100%;
  height: 100%;
  background-color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  border: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 100%;
}

.c1:hover {
  border: 0;
}

.c3 {
  height: auto;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 10px 0px 20px;
  width: 100%;
}

.c6 h3 {
  margin: 0;
}

@media (min-width:740px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:740px) {
  .c3 {
    width: 45%;
  }
}

@media (min-width:1440px) {
  .c3 {
    width: 100%;
  }
}

<article
  className="c0"
>
  <a
    className="c1 c2"
    color="red"
    href="/test"
    target="_self"
    type="standard"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
        height="auto"
        width="100%"
      >
        <img
          alt="Image's description"
          className="lazyload c5"
          data-src="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
          height="auto"
          width="100%"
        />
      </div>
    </div>
    <div
      className="c6"
    >
      <span
        className="c7"
        color="inherit"
        size="xs"
      >
         
      </span>
      <h3
        className="c8"
        color="inherit"
        size="xl"
      >
        News article
      </h3>
      
    </div>
  </a>
</article>
`;

exports[`renders correctly with copy 1`] = `
.c7 {
  font-size: 0.83125rem;
  text-transform: uppercase;
}

.c8 {
  font-size: 1.725rem;
  text-transform: uppercase;
}

.c9 {
  font-size: 1.2rem;
  text-transform: inherit;
}

.c4 {
  position: relative;
  width: 100%;
  height: auto;
}

.c5 {
  display: block;
  width: 100%;
  height: auto;
  object-fit: cover;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0 2px 1px;
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: normal;
}

.c2:hover {
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: 700;
}

.c0 {
  width: 100%;
  height: 100%;
  background-color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  border: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 100%;
}

.c1:hover {
  border: 0;
}

.c3 {
  height: auto;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 10px 0px 20px;
  width: 100%;
}

.c6 h3 {
  margin: 0;
}

@media (min-width:740px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:740px) {
  .c3 {
    width: 45%;
  }
}

@media (min-width:1440px) {
  .c3 {
    width: 100%;
  }
}

<article
  className="c0"
>
  <a
    className="c1 c2"
    color="red"
    href="/test"
    target="_self"
    type="standard"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
        height="auto"
        width="100%"
      >
        <img
          alt="Image's description"
          className="lazyload c5"
          data-src="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
          height="auto"
          width="100%"
        />
      </div>
    </div>
    <div
      className="c6"
    >
      <span
        className="c7"
        color="inherit"
        size="xs"
      >
         
      </span>
      <h3
        className="c8"
        color="inherit"
        size="xl"
      >
        News article
      </h3>
      <p
        className="c9"
        color="inherit"
        size="m"
      >
        News article copy
      </p>
    </div>
  </a>
</article>
`;

exports[`renders correctly with date 1`] = `
.c7 {
  font-size: 0.83125rem;
  text-transform: uppercase;
}

.c8 {
  font-size: 1.725rem;
  text-transform: uppercase;
}

.c4 {
  position: relative;
  width: 100%;
  height: auto;
}

.c5 {
  display: block;
  width: 100%;
  height: auto;
  object-fit: cover;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0 2px 1px;
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: normal;
}

.c2:hover {
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: 700;
}

.c0 {
  width: 100%;
  height: 100%;
  background-color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  border: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 100%;
}

.c1:hover {
  border: 0;
}

.c3 {
  height: auto;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 10px 0px 20px;
  width: 100%;
}

.c6 h3 {
  margin: 0;
}

@media (min-width:740px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:740px) {
  .c3 {
    width: 45%;
  }
}

@media (min-width:1440px) {
  .c3 {
    width: 100%;
  }
}

<article
  className="c0"
>
  <a
    className="c1 c2"
    color="red"
    href="/test"
    target="_self"
    type="standard"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
        height="auto"
        width="100%"
      >
        <img
          alt="Image's description"
          className="lazyload c5"
          data-src="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
          height="auto"
          width="100%"
        />
      </div>
    </div>
    <div
      className="c6"
    >
      <span
        className="c7"
        color="inherit"
        size="xs"
      >
         01 July 2019
      </span>
      <h3
        className="c8"
        color="inherit"
        size="xl"
      >
        News article
      </h3>
      
    </div>
  </a>
</article>
`;

exports[`renders correctly with date and type 1`] = `
.c7 {
  font-size: 0.83125rem;
  text-transform: uppercase;
}

.c8 {
  font-size: 1.725rem;
  text-transform: uppercase;
}

.c4 {
  position: relative;
  width: 100%;
  height: auto;
}

.c5 {
  display: block;
  width: 100%;
  height: auto;
  object-fit: cover;
}

.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  padding: 0 2px 1px;
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: normal;
}

.c2:hover {
  color: #2C0230;
  border-bottom: 2px solid #2C0230;
  font-weight: 700;
}

.c0 {
  width: 100%;
  height: 100%;
  background-color: #fff;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  border: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 100%;
}

.c1:hover {
  border: 0;
}

.c3 {
  height: auto;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0px 10px 0px 20px;
  width: 100%;
}

.c6 h3 {
  margin: 0;
}

@media (min-width:740px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

@media (min-width:740px) {
  .c3 {
    width: 45%;
  }
}

@media (min-width:1440px) {
  .c3 {
    width: 100%;
  }
}

<article
  className="c0"
>
  <a
    className="c1 c2"
    color="red"
    href="/test"
    target="_self"
    type="standard"
  >
    <div
      className="c3"
    >
      <div
        className="c4"
        height="auto"
        width="100%"
      >
        <img
          alt="Image's description"
          className="lazyload c5"
          data-src="http://images.ctfassets.net/zsfivwzfgl3t/Yq59XdwwQgjNOxky93K1Q/17c2d80dce99067b0b3508f33075cbe3/funding_4-3_2x.jpg"
          height="auto"
          width="100%"
        />
      </div>
    </div>
    <div
      className="c6"
    >
      <span
        className="c7"
        color="inherit"
        size="xs"
      >
         article | 01 July 2019
      </span>
      <h3
        className="c8"
        color="inherit"
        size="xl"
      >
        News article
      </h3>
      
    </div>
  </a>
</article>
`;
